src/pip/_internal/index.py

```diff
@@ -56,6 +56,8 @@
 
 logger = logging.getLogger(__name__)
 
+blacklisted_hosts = set()
+
```

Comment:
i wonder if there is a better place than a global for this variable

Reply:
Yes, open to suggestions on this one. It could be passed through all the classes. Because of its functionality / purpose, I'm not sure that this is something that needs, for example, to be multiprocess-safe.